### PR TITLE
feat(utxo-bin): show spends of type HollowSegwitSpend

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
Show "Hollow Segwit Spends" - spends of wrapped segwit output scripts on
non-segwit chains.

See https://bitcoin.stackexchange.com/a/115850/137601 for further details.

Issue: BG-60311

### CODE REVIEW

**Comment by OttoAllmendinger** [comment]
Sample output

```
│ ├─┬ 0: b504dc934684d2633747cf7ac9d12fadd6db96278562124e180bcd01b18f44ba:0
│ │ ├── sequence: ffffffff
│ │ ├─┬ scriptSig: scripthash
│ │ │ └─┬ 0: 0014745173b196633c6d2dbce5f3a8f9021d6b109f11
│ │ │   └── asm: 
│ │ │       OP_0
│ │ │       745173b196633c6d2dbce5f3a8f9021d6b109f11
│ │ ├── witness: []
│ │ ├─┬ sigScript: p2shP2wpkhHollow
│ │ │ └── info: This output was likely claimed by a miner. See https://bitcoin.stackexchange.com/a/115850/137601 for details.
│ │ ├── value: 0.00042968
│ │ └─┬ pubScript: a91472fc0f354476553fff91b7ecf99c434de06c50a487
│ │   └── address: 3CAzuom4jahfwgptXPpLKeNH2XZMoJkBMW
```

File: modules/utxo-bin/src/hollowSegwitSpend.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+export function parseHollowSegwitSpend(
+  input: utxolib.TxInput,
+  network: utxolib.Network
+): HollowSegwitSpend | undefined {
+  if (
+    utxolib.getMainnet(network) !== utxolib.networks.bitcoincash &&
+    utxolib.getMainnet(network) !== utxolib.networks.bitcoinsv
+  ) {
+    // this can only happen on the BCH/BSV forks
```
Comment (sangaman) [inline modules/utxo-bin/src/hollowSegwitSpend.ts:30]: BSV even did away with P2SH I believe, so there can't be new P2SH outputs created. Although I suppose it's possible that there are old "hollow" outputs out there.

**Review by sangaman — APPROVED** [review]
Excellent stack overflow answer and code looks great.

File: modules/utxo-bin/src/hollowSegwitSpend.ts (hunk 1 above)
Comment (OttoAllmendinger) [reply modules/utxo-bin/src/hollowSegwitSpend.ts:30]: right I remember now

**Pull request merged**